Update query syntax docs (#3671)
weaveworks/weave-gitops: website/docs/explorer/querying.mdx

|------------| ---------| ----------------|---------|----------|------------------------|
|`default/progress-delivery-demo2-32` | `flux-system`   | `HelmRelease` | `podinfo` | `Success` | `Release reconciliation succeeded`  |

> And represented in the UI like

Reviewer: it seems that we miss a new header between schema and showing filters (or some other content that links both together) 

Reading it as currently is feels disconnected 

![Screenshot 2023-05-31 at 10 25 33](https://github.com/weaveworks/weave-gitops/assets/12957664/32347896-ba84-4906-9fd5-1ed1366b5007)

> ![explorer](imgs/explorer-filter-open.png)

Reviewer: We could reduce this screenshoot (may be just showing two categories ) ... we could have the same info with less space.

![explorer](imgs/explorer-query-overview.png)

## Query Syntax

> The Explorer query system uses the[Bleve](https://blevesearch.com) search indexer and the [query language](https://blevesearch.com/docs/Query-String-Query/) that accompanies it.

Reviewer 2: ```suggestion
The Explorer query system uses the [Bleve](https://blevesearch.com) search indexer and the [query language](https://blevesearch.com/docs/Query-String-Query/) that accompanies it.
```

A brief introduction to the query syntax:

Simple string matching query will match on all fields:
```
podinfo
```

To add field scoping to a query, use a `:` between key and value:
```
kind:Kustomization
```

> To add required fields, use the `+` operator. Use the `-` to exclude certain fields. This will return all kustomizations NOT on the management cluster:

Reviewer 2: ```suggestion
To make a term mandatory, put `+` in front. Use `-` in front of a term to exclude matches from the result. This will return all kustomizations NOT on the management cluster:
```

This is my thinking: it's the term (field and value, or value in any field, to be matched) that's required or excluded.

Author: No longer relevant after moving to faceted search

```
+kind:Kustomization -cluster:management
```

> Regular expressions are also suported:

Reviewer 2: ```suggestion
Regular expressions are also supported, by enclosing in slashes `/`:
```
